src/ray/raylet/lineage_cache.cc

```diff
@@ -335,8 +335,11 @@ Lineage LineageCache::GetUncommittedLineage(const TaskID &task_id,
   // already explicitly forwarded to this node before.
   if (uncommitted_lineage.GetEntries().empty()) {
     auto entry = lineage_.GetEntry(task_id);
-    RAY_CHECK(entry);
-    RAY_CHECK(uncommitted_lineage.SetEntry(entry->TaskData(), entry->GetStatus()));
+    if (entry) {
+      RAY_CHECK(uncommitted_lineage.SetEntry(entry->TaskData(), entry->GetStatus()));
+    } else {
+      RAY_LOG(ERROR) << "No lineage cache entry found for task " << task_id;
```

Comment:
It would be nice to add a comment on under what conditions this entry doesn't exist.

Reply:
Done. Added a pointer back to the issue.

It's not super clear what is evicting the lineage: perhaps some race condition on a task getting rescheduled, or the task succeeding but the node failing after that.